backend/resource2.py

```diff
@@ -300,21 +303,33 @@ def overwrite_JH_with_file_from_HS(self, filepath):
         self.remote_folder.download_file_to_JH(filepath, self.path_prefix)
         self.JH_files = self.get_files_upon_init_JH()
 
-    def overwrite_HS_with_file_from_JH(self, filepath):
+    def overwrite_HS_with_file_from_JH(self, file_path):
         """ overwrites HS file with one from JH """
-        if filepath is not None:
-            pathWithoutType, fileType = filepath.split(".")
-            if self.is_file_in_HS(pathWithoutType, fileType):
-                self.delete_file_or_folder_from_HS(filepath)
-            elif "/" in filepath:
-                folderPath = filepath.rsplit('/', 1)[0]
-                if self.is_folder_in_HS(folderPath) == False:
-                    self.remote_folder.create_folder(folderPath)
-
-            self.remote_folder.upload_file_to_HS(self.path_prefix+filepath, filepath)
-            self.hs_files = self.get_files_upon_init_HS()
-        else:
-            logging.info('Missing filepath in call to overwrite')
+        if str(file_path).startswith('/'):
+            file_path = str(file_path)[1:]
+        full_file_path_rel_resource_root = Path(file_path)
+        file_extension = full_file_path_rel_resource_root.suffix
+        path_without_extension = str(full_file_path_rel_resource_root)[:-len(file_extension)]
+        # Drop the leading . from the file extension
+        file_extension = file_extension[1:]
+        if self.is_file_or_folder_in_HS(path_without_extension, file_extension):
+            try:
+                self.delete_file_or_folder_from_HS(full_file_path_rel_resource_root)
+            except hs_restclient.HydroShareHTTPException as e:
```

Comment:
what does the hydroshare exception trigger?

Reply:
Actually I should remove those `try...except` blocks.